test/framework/cluster_proxy.go

```diff
@@ -226,12 +228,24 @@ func (p *clusterProxy) CollectWorkloadClusterLogs(ctx context.Context, namespace
 
 	for i := range machines.Items {
 		m := &machines.Items[i]
-		err := p.logCollector.CollectMachineLog(ctx, p.GetClient(), m, path.Join(outputPath, m.GetName()))
+		err := p.logCollector.CollectMachineLog(ctx, p.GetClient(), m, path.Join(outputPath, "machines", m.GetName()))
 		if err != nil {
 			// NB. we are treating failures in collecting logs as a non blocking operation (best effort)
 			fmt.Printf("Failed to get logs for machine %s, cluster %s/%s: %v\n", m.GetName(), namespace, name, err)
 		}
 	}
+
+	machinePools, err := getMachinePoolsInCluster(ctx, p.GetClient(), namespace, name)
+	Expect(err).ToNot(HaveOccurred(), "Failed to get machine pools for the %s/%s cluster", namespace, name)
+
+	for i := range machinePools.Items {
+		mp := &machinePools.Items[i]
+		err := p.logCollector.CollectMachinePoolLog(ctx, p.GetClient(), mp, path.Join(outputPath, "machine-pools", mp.GetName()))
+		if err != nil {
+			// NB. we are treating failures in collecting logs as a non blocking operation (best effort)
+			fmt.Printf("Failed to get logs for machine pool %s, cluster %s/%s: %v\n", mp.GetName(), namespace, name, err)
```

Comment:
nit: can render the namespace/name in one shot with `(%v, client.ObjectKeyFromObject(mp))`

Reply:
this is the cluster name / namespace, not the machinepool's (and we don't have the cluster object)